src/blockchains/eth/lib/custom_trie.ts

```diff
@@ -0,0 +1,152 @@
+
+import { assertIsDefined } from '../../../lib/utils';
+import { Trace } from '../eth_types';
+
+
+export class TrieNode {
```

Comment:
This implementation is not brand new, despite being shown as a new file in the PR. It is just migrated to TS.